homeassistant/components/nextcloud/config_flow.py

```diff
@@ -79,3 +101,45 @@ async def async_step_user(
         return self.async_show_form(
             step_id="user", data_schema=data_schema, errors=errors
         )
+
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle flow upon an API authentication error."""
+        self._entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        self._url = entry_data[CONF_URL]
+        self._username = entry_data[CONF_USERNAME]
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle reauthorization flow."""
+        errors = {}
+
+        if user_input is not None:
+            assert self._entry is not None
+            try:
+                await self.hass.async_add_executor_job(
+                    self._try_connect_nc, {**self._entry.data, **user_input}
+                )
+            except NextcloudMonitorAuthorizationError:
+                errors["base"] = "invalid_auth"
+            except (NextcloudMonitorConnectionError, NextcloudMonitorRequestError):
+                errors["base"] = "connection_error"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self._entry,
+                    data={**self._entry.data, **user_input},
+                )
+                await self.hass.config_entries.async_reload(self._entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        data_schema = self.add_suggested_values_to_schema(
+            DATA_SCHEMA_REAUTH,
+            user_input,
```

Comment:
Does this work to have the initial username?
```suggestion
            {CONF_USERNAME: self._entry.data[CONF_USERNAME], **user_input},
```

Reply:
unfortunately not:
```
        data_schema = self.add_suggested_values_to_schema(
            DATA_SCHEMA_REAUTH,
>           {CONF_USERNAME: self._entry.data[CONF_USERNAME], **user_input},
        )
E       TypeError: 'NoneType' object is not a mapping
```

but this works
```python
        data_schema = self.add_suggested_values_to_schema(
            vol.Schema(
                {
                    vol.Required(
                        CONF_USERNAME, default=self._entry.data[CONF_USERNAME]
                    ): str,
                    vol.Required(CONF_PASSWORD): str,
                }
            ),
            user_input,
        )
```

Reply:
I don't like default ! I'm running into all sorts of issues with it.
Maybe this works?
`suggested_value = {CONF_USERNAME: self._entry.data[CONF_USERNAME], **(user_input or {})}`

Reply:
this works 👍